portable/NetworkInterface/Common/phyHandling.c

```diff
@@ -234,6 +234,12 @@ BaseType_t xPhyDiscover( EthernetPhy_t * pxPhyObject )
             pxPhyObject->fnPhyRead( xPhyAddress, phyREG_02_PHYSID1, &ulUpperID );
             ulPhyID = ( ( ( uint32_t ) ulUpperID ) << 16 ) | ( ulLowerID & 0xFFF0U );
 
+            /* KSZ8795 device index starts at 1, skip registering the 0 phyAddress */
+            if ( ulPhyID == PHY_ID_KSZ8795  && xPhyAddress == 0U )
+            {
+                continue;
+            }

```

Comment:
@ancantus, thanks for this PR.
But just curious, what happens if you do not put this continue statement here?
Why don't you skip this entry later on in your code? Wouldn't it have importance for other users?

Reply:
If I don't skip the 0 `xPhyAddress` then the phyHandling will only bring up ports 1 - 3 and the 4th port of the switch would not be activated (when ipconfigPHY_MAX_PORTS == 4). This is because for some reason the KSZ8795 switch's ports start at `xPhyAddress == 1`.

Reply:
I'm not sure what you mean by having importance for other users. By my reckoning this is where the mapping for `ucPhyIndexes` to `xPhyAddress` is done so the logical ports match up to the physical addresses on the MIIM. If someone's using a KSZ8795 switch they're probably going to want to address all 4 ports. 

But very interested in your advice if I've done something that might mess up other users here. Your way more familiar with this codebase and other uses than I am!

Reply:
I haven't used a KSZ8795 switch yet. The datasheet says it's a 5-port switch:

"Four Internal Media Access Control (MAC1 to MAC4) Units and One Internal Gigabit Media Access Control (GMAC5) Unit"

And it looks like they are mapped on addresses 0 up to 4.

What happens if you define ipconfigPHY_MAX_PORTS as 5?
Can you try that out?

Do you have a board with this PHY?
Is the PHY connect to 4 connectors?

What platform (CPU) are you using? Does Ethernet work properly?

Reply:
@ancantus thanks for this PR.
Do you have any update on these questions?

Thanks,
Aniruddha

Reply:
Sorry guys! I've been busy with other work tasks this week.

@htibosch: You pointed out some good stuff, I'll double check everything next week on my board that does have this switch on it. The 5th port is (in my layout) used to connect to the CPU via RMII so is not configured like the others (in my understanding).  The other 4 ports are connected to external connectors.

We're using a STM32H743 and with these changes Ethernet works properly. Without these changes there are issues (which I'll post next week with a more details).

Reply:
I fully retested today and the phyHandling code does fully work if the `ipconfigPHY_MAX_PORTS` is set to 5. I think for initial porting work I was connected to the 5th port and missed that the RMII port is the 0th. 

I've updated the PR to remove that offset code, but kept the ID. I'm also fine with just closing this PR and not merging as there's really no special work required here to support the switch. I'm not sure what's your policy for supporting new PHY's.

Reply:
Through code reviewing this change to my company's fork I realized I had missed the other change required for this switch. The 0x1F MIIM register's bit fields for port operation mode indicators is different than what's coded. As documented in `phyHandling.c`:
```
/* Bit fields for 'phyREG_1F_PHYSPCS
 *  001 = 10BASE-T half-duplex
 *  101 = 10BASE-T full-duplex
 *  010 = 100BASE-TX half-duplex
 *  110 = 100BASE-TX full-duplex
 */
```
Where the KSZ8795 datasheet has:
 010 = 10BASE-T half-duplex
 101 = 10BASE-T full-duplex
 011 = 100BASE-TX half-duplex
 110 = 100BASE-TX full-duplex
Screenshot of the section of the manual below. 
![image](https://user-images.githubusercontent.com/1694065/158288435-bdf978df-51f2-4abe-977c-fc7474da1872.png)

Change should be up in a few mins. My apologies for the churn on this PR. I did the work on this a little while ago in our internal fork & didn't keep track of the changes as much as I should have.